python/lsst/qa/explorer/classifyObjects.py

```diff
@@ -0,0 +1,371 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Applys star galaxy classifier
+"""
+
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import Task
+from .parquetTable import ParquetTable
+
+import numpy as np
+import pickle
+
+
+class StarGalaxyClassifierConfig(pexConfig.Config):
+    """Config for star galaxy classifier
+    """
+    version = pexConfig.Field(
+        doc="Version information",
+        dtype=str,
+        default="HSC-COSMOS-20180528",
+    )
+
+    pass
+
+
+class StarGalaxyClassifierTask(Task):
+    """Add classification (star or galaxy) and associated flag columns to a parquet table.
+    """
+
+    _DefaultName = "starGalaxyClassifier"
+    ConfigClass = StarGalaxyClassifierConfig
+
+    def __init__(self):
+        Task.__init__(self)
+
+    def runDataRef(self, dataRef):
+        """Add required magnitudes and run the classifier
+
+        checks that all the required filters are in the file, then adds magntiudes and runs the classifier
+        for the given patch and tract.
+        """
+        cat = dataRef.get("deepCoadd_obj").toDataFrame({"dataset": "meas"})
+
+        # For now raise a not implemented error for other cameras, eventually needs other trained classifiers
+        camera_name = dataRef.getButler()._getDefaultMapper().getCameraName()
+        if camera_name != "hsc":
+            raise NotImplementedError
+
+        # To Do: DM-14539 - Move this to somewhere else
+        # put this into /datasets/hsc then the butler will get it
+        # Add classifier_pickle to obs_base datasets, copy deepCoadd_skyMap.
+        clfFilenameMorph = "/home/sr525/starGalaxy/clfIsotonicMorph.pickle"
+        with open(clfFilenameMorph, "rb") as f:
+            filters = pickle.load(f)
+
+        filtersInCat = set(cat.columns.get_level_values(0))
+        if filtersInCat != set(filters):
+            raise RuntimeError("Not all filters required in file")
+
+        # addMagnitdues will probably be removed later as it will hopefully be superseded by Tim
+        cat = addMagnitudes(cat, dataRef, filters)
+        cat = self.run(cat, filters)
+
+        cat = ParquetTable(dataFrame=cat)
+        dataRef.put(cat, "deepCoadd_sg")
+
+    def run(self, cat, filters):
+        """Add colour based classification, morphology only based classification and their associated flags.
```

Comment:
It is surprising and disconcerting that `runDataRef` and `run` have such dramatically different doc strings. Normally `runDataRef` just unpacks the data in a data ref and calls `run`. Then the docs for `runDataRef` can say pretty much just that, including information about what kind of data ref is required.